application/constants/templates.py

```diff
@@ -43,3 +49,51 @@ class HookTypes(StrEnum):
     Template application hook types.
     """
     kubernetes_job = 'kubernetes_job'
+
+
+class http(StrEnum):
+    """
+    Template HTTP component types.
+    """
+    url = 'url'
+    method = 'method'
+    headers = 'headers'
+    parameters = 'parameters'
+
+class httpMethods(StrEnum):
+    """
+    Template HTTP component methods.
+    """
+    get = 'GET'
+    post = 'POST'
+    put = 'PUT'
+    patch = 'PATCH'
+    delete = 'DELETE'
+
+
+class httpDeploy(BaseModel):
+    """
+    Deploy schema for HTTP type components.
+    """
+    url: constr(min_length=1, strip_whitespace=True) = Field(description='URL for HTTP deployment', example='http://example.com')
+    method: Union[httpMethods, constr(min_length=1, strip_whitespace=True)] = Field(default=httpMethods.get, description='HTTP method for deployment', example=httpMethods.post)
+    headers: Optional[dict] = Field(description='Headers for HTTP deployment', example={'Content-Type': 'application/json'})
+    parameters: Optional[dict | list[dict]] = Field(description='Parameters for HTTP deployment', example=[{'param1': 'value1'}])
```

Comment:
Modern type notation. `Optional[dict | list[dict]]` can be replaced by `dict | list[dict] | None`. 

But! I think it can be replaced by `dict[str, str]`.

1. Same reason as for `headers`. What is the point of supporting 2 values for representing same meaning(`None` and `{}`)?
2. What is the point of supporting format `[{'param1': 'value1'}]`? How you will handle case when user pass:
```
[
    {'param1': 'value1'},
    {'param1': 'value2'}
]
```
?

Reply:
This is to be able to support both YAML list and a map, ie - the user can do
```
parameters:
- key1: value1
- key2: values2
```
or
```
parameters:
  key1: value1
  key2: values2
```